rts/motoko-rts/src/types.rs

```diff
@@ -598,17 +570,23 @@ impl Region {
 #[repr(C)] // See the note at the beginning of this module
 pub struct Object {
     pub header: Obj,
-    pub size: u32,     // Number of elements
-    pub hash_ptr: u32, // Pointer to static information about object field labels. Not important for GC (does not contain pointers).
+    pub hash_blob: Value, // Pointer to a blob containing the hashes of the object field labels.
```

Comment:
Do we still share these across all objects with the same fields? That should be possible with your new object pooling, right?

Reply:
Good point: I implemented https://github.com/dfinity/motoko/pull/4492 as additional optimization to share the hash blob for repeated dynamic allocation of the same typed-object.

Reply:
Btw: The sharing does not yet apply to different static occurrences of same-structured object. I.e. it is only effective if the same "code line of allocation" gets executed multiple times.